apps/search_providers/service.py

```diff
@@ -41,6 +43,23 @@ def find_one(self, req, **lookup):
         provider = super().find_one(req, **lookup)
         return provider if provider and provider['search_provider'] in allowed_search_providers else None
 
+    def on_created(self, docs):
+        for doc in docs:
+            if doc.get('is_default'):
+                self.find_and_modify(
```

Comment:
This can be extracted to a separate function as it is used in on_updated as well...